src/charm.py

```diff
@@ -458,11 +458,17 @@ def configure_app_leader(self, event: ops.framework.EventBase) -> None:
     def configure_app_non_leader(self, event: ops.framework.EventBase) -> None:
         """Configure the non leader unit."""
         super().configure_app_non_leader(event)
+
+        # MicroClusterNodeAddedEvent triggered only when token is present.
         if isinstance(event, MicroClusterNodeAddedEvent):
             self.cluster_nodes.join_node_to_cluster(event)
 
-        if self.peers.interface.state.joined:
-            self.manage_rgw_service(event)
+        if not self.peers.interface.state.joined:
+            # deferral not needed as join token is not yet received.
```

Comment:
Suggest to add info logging here

Reply:
Since this is a sunbeam guard exception, it will always generate a log.

Reply:
exception would result in a log entry.